components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParAuthServiceImpl.java

```diff
@@ -18,81 +18,106 @@
 
         return parAuthResponse;
     }
 
-    private void persistParRequest(String uuid, Map<String, String> params, long scheduledExpiryTime)
+    private void persistParRequest(String uuid, Map<String, String> params, long expiresIn)
             throws ParCoreException {
 
-        parMgtDAO.persistParRequest(uuid, params.get(OAuthConstants.OAuth20Params.CLIENT_ID),
-                scheduledExpiryTime, params);
+        parMgtDAO.persistRequestData(uuid, params.get(OAuthConstants.OAuth20Params.CLIENT_ID),
+                expiresIn, params);
     }
 
+    @Override
     public Map<String, String> retrieveParams(String uuid, String clientId) throws ParCoreException {
 
-        ParRequestDO parRequestDO = parMgtDAO.getParRequest(uuid);
-        parMgtDAO.removeParRequest(uuid);
-        isRequestUriExpired(parRequestDO.getScheduledExpiryTime());
-        isClientIdValid(clientId, parRequestDO.getClientId());
+        ParRequestDO parRequestDO = parMgtDAO.getRequestData(uuid);
+        parMgtDAO.removeRequestData(uuid);
+        validateRequestURI(parRequestDO.getExpiresIn());
+        validateClientID(clientId, parRequestDO.getClientId());
 
         return parRequestDO.getParams();
     }
 
-    private void isRequestUriExpired(long scheduledExpiryTime) throws ParCoreException {
+    private void validateRequestURI(long expiresIn) throws ParCoreException {
 
         long currentTimeInMillis = Calendar.getInstance(TimeZone.getTimeZone(ParConstants.UTC)).getTimeInMillis();
 
-        if (currentTimeInMillis > scheduledExpiryTime) {
+        if (currentTimeInMillis > expiresIn) {
             throw new ParCoreException(OAuth2ErrorCodes.INVALID_REQUEST, "request_uri expired");
         }
     }
 
-    private void isClientIdValid(String clientId, String parClientId) throws
+    private void validateClientID(String clientId, String parClientId) throws
             ParCoreException {
 
         if (!StringUtils.equals(parClientId, clientId)) {
-            throw new ParCoreException(OAuth2ErrorCodes.INVALID_CLIENT, "client_ids does not match");
+            throw new ParCoreException(OAuth2ErrorCodes.INVALID_CLIENT, "client_ids do not match.");
+        }
+    }
+
+    private static long getExpiresInValue() throws ParCoreException {
+
+        try {
+            OMElement parConfig = IdentityConfigParser.getInstance().getConfigElement(ParConstants.CONFIG_ELEM_OAUTH)
+                    .getFirstChildWithName(new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE,
+                            ParConstants.PAR));
+            if (parConfig != null) {
+                String expiryTimeValue = parConfig.getFirstChildWithName(new QName(
+                        IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, ParConstants.EXPIRY_TIME)).getText();
+                if (StringUtils.isNotBlank(expiryTimeValue)) {
+                    return Long.parseLong(expiryTimeValue);
+                }
+            }
```

Comment:
let's add a debug log here saying that we are setting the expiry time to a default value